physics/discrete_trajectory_body.hpp

```diff
@@ -311,19 +318,37 @@ Instant DiscreteTrajectory<Frame>::t_max() const {
 template<typename Frame>
 Position<Frame> DiscreteTrajectory<Frame>::EvaluatePosition(
     Instant const& time) const {
-  return GetInterpolation(time).Evaluate(time);
+  auto const iter = this->LowerBound(time);
+  if (iter->time == time) {
+    return iter->degrees_of_freedom.position();
+  }
+  CHECK_LE(t_min(), time);
```

Comment:
I believe that these should be `CHECK_LT` and `CHECK_GT` because the equality case would have returned at line 323.  Same in the other functions.